File: 1. Creational Design Patterns/Prototype/C#/PrototypeTemplate.cs
// Prototype Pattern – C# Template
using System;
namespace PrototypeTemplate
{
    interface IPrototype<T>
    {
        T Clone();
    }
    class ConcretePrototype : IPrototype<ConcretePrototype>
    {
        public int Data;
        public ConcretePrototype Clone() => (ConcretePrototype) MemberwiseClone();
    }
    class Program{ static void Main(){ var original=new ConcretePrototype{Data=42}; var copy=original.Clone(); }}
}

File: 1. Creational Design Patterns/Prototype/Go/PrototypeTemplate.go
package main

// Prototype Pattern – Go Template

type Prototype interface {
	Clone() Prototype
}

type ConcretePrototype struct{ Data int }
func (p *ConcretePrototype) Clone() Prototype { cp := *p; return &cp }

func main(){ original:=&ConcretePrototype{Data:42}; copy:=original.Clone(); _ = copy }
